File: Java/기본 수학2/터렛.java
import java.io.*;
import java.util.StringTokenizer;
import java.math.*;

public class Main{
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		for(int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int x1 = Integer.parseInt(st.nextToken()); int y1 = Integer.parseInt(st.nextToken());
			int r1 = Integer.parseInt(st.nextToken()); int x2 = Integer.parseInt(st.nextToken());
			int y2 = Integer.parseInt(st.nextToken()); int r2 = Integer.parseInt(st.nextToken());
			double d = Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1,2));
			int r_1 = r1 + r2;
			int r_2 = Math.abs(r1 - r2);

			if(d == 0) {
				if(r1 == r2) System.out.println(-1);
				else System.out.println(0);
			}
			else {
				if(d == r_1 || d == r_2) System.out.println(1);
				else if(d < r_1 && d > r_2) System.out.println(2);
			    else System.out.println(0);
			}
		}
    }
}

File: C++/기본 수학2/터렛.cc
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int t, x[2], y[2], r[2];
    cin >> t;
    for(int i = 0; i < t; i++) {
        cin >> x[0] >> y[0] >> r[0] >> x[1] >> y[1] >> r[1];
    
        double d = sqrt(pow(x[1] - x[0], 2) + pow(y[1] - y[0],2));
		int r_1 = abs(r[1] - r[0]);
		int r_2 = r[1] + r[0];

		if(d == 0) {
			if(r[1] == r[0])
			    cout << -1 << '\n';
			else
			    cout << 0 << '\n';
		}
		else {
			if(d == r_1 || d == r_2)
			    cout << 1 << '\n';
			else if(r_1 < d && d < r_2)
    			cout << 2 << '\n';
			else
			    cout << 0 << '\n';
		}
    }
    
    return 0;
}